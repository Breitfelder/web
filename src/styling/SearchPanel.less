.bblfsh-search-panel {
    &__container {
        padding: 0.5rem 1rem;
        display: flex;
        background: @background-light;
        border-bottom: 1px solid @border-smooth;
        min-height: 2.5rem;
    }

    &__field {
        display: flex;
        align-items: center;

        & + & {
            margin-left: 1rem;
        }
    }

    &__text-input {
        width: 100%;
        background: transparent;
        border-top: none;
        border-left: none;
        border-right: none;
        border-bottom: 1px solid @border-smooth;
        outline: none;
        font-size: 0.8rem;
        padding: 0.1rem 0.5rem;
    }

    &__button {
        .bblfsh-button();
        border: none;
    }
}
