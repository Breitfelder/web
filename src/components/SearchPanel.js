import React from 'react';
import styled from 'styled-components';
import { background, border } from '../styling/variables';
import { CssButton } from './Button';
import { connect } from 'react-redux';
import { setUastQuery } from 'state/options';
import { runParserWithQuery } from 'state/code';

const Container = styled.form`
  padding: 0.5rem 1rem;
  display: flex;
  background: ${background.light};
  border-bottom: 1px solid ${border.smooth};
  min-height: 2.5rem;
`;

const Field = styled.div`
  display: flex;
  align-items: center;

  & + & {
    margin-left: 1rem;
  }
`;

const TextInput = styled.input`
  width: 100%;
  background: transparent;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: 1px solid ${border.smooth};
  outline: none;
  font-size: 0.8rem;
  padding: 0.1rem 0.5rem;
`;

const Button = styled.input`
  ${CssButton} border: none;
`;

export function SearchPanel({ query, loading, onChange, onSearch }) {
  const onSubmit = e => {
    e.preventDefault();
    if (loading) {
      return;
    }
    onSearch();
  };

  return (
    <Container onSubmit={onSubmit}>
      <Field style={{ width: '100%' }}>
        <TextInput
          type="text"
          name="query"
          value={query}
          onChange={e => onChange(e.target.value)}
          disabled={loading}
          placeholder="UAST Query"
        />
      </Field>
      <Field>
        <Button type="submit" value="Search" disabled={loading || !query} />
      </Field>
      <Field>
        <a
          href="https://doc.bblf.sh/user/uast-querying.html"
          target="_blank"
          rel="noopener noreferrer"
        >
          Help
        </a>
      </Field>
    </Container>
  );
}

const mapStateToProps = state => {
  return {
    query: state.options.uastQuery,
    loading: state.code.loading,
    isValid: true,
  };
};

const mapDispatchToProps = {
  onChange: setUastQuery,
  onSearch: runParserWithQuery,
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchPanel);
