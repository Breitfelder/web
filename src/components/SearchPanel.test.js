import React from 'react';
import renderer from 'react-test-renderer';
import { SearchPanel } from './SearchPanel';

const shouldMatchSnapshot = comp => {
  const wrapper = renderer.create(comp);
  expect(wrapper.toJSON()).toMatchSnapshot();
};

it('renders correctly', () => {
  shouldMatchSnapshot(<SearchPanel />);
});

it('renders correctly with value', () => {
  shouldMatchSnapshot(<SearchPanel query="test" />);
});

it('input & button should be disabled while loading', () => {
  shouldMatchSnapshot(<SearchPanel loading={true} query="test" />);
});
