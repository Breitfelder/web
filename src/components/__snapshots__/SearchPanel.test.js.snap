// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input & button should be disabled while loading 1`] = `
.c0 {
  padding: 0.5rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #f9f7f5;
  border-bottom: 1px solid #dad7d0;
  min-height: 2.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 + .c1 {
  margin-left: 1rem;
}

.c2 {
  width: 100%;
  background: transparent;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: 1px solid #dad7d0;
  outline: none;
  font-size: 0.8rem;
  padding: 0.1rem 0.5rem;
}

.c3 {
  padding: 0.5em 0.5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  text-transform: uppercase;
  font-size: 0.8rem;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  border: none;
}

.c3[disabled] {
  opacity: 0.6;
  pointer-events: none;
}

.c3:hover {
  box-shadow: 0 5px 15px rgba(138,128,115,0.8);
}

<form
  className="c0"
  onSubmit={[Function]}
>
  <div
    className="c1"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <input
      className="c2"
      disabled={true}
      name="query"
      onChange={[Function]}
      placeholder="UAST Query"
      type="text"
      value="test"
    />
  </div>
  <div
    className="c1"
  >
    <input
      className="c3"
      disabled={true}
      type="submit"
      value="Search"
    />
  </div>
  <div
    className="c1"
  >
    <a
      href="https://doc.bblf.sh/using-babelfish/uast-querying.html"
      rel="noopener noreferrer"
      target="_blank"
    >
      Help
    </a>
  </div>
</form>
`;

exports[`renders correctly 1`] = `
.c0 {
  padding: 0.5rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #f9f7f5;
  border-bottom: 1px solid #dad7d0;
  min-height: 2.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 + .c1 {
  margin-left: 1rem;
}

.c2 {
  width: 100%;
  background: transparent;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: 1px solid #dad7d0;
  outline: none;
  font-size: 0.8rem;
  padding: 0.1rem 0.5rem;
}

.c3 {
  padding: 0.5em 0.5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  text-transform: uppercase;
  font-size: 0.8rem;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  border: none;
}

.c3[disabled] {
  opacity: 0.6;
  pointer-events: none;
}

.c3:hover {
  box-shadow: 0 5px 15px rgba(138,128,115,0.8);
}

<form
  className="c0"
  onSubmit={[Function]}
>
  <div
    className="c1"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <input
      className="c2"
      disabled={undefined}
      name="query"
      onChange={[Function]}
      placeholder="UAST Query"
      type="text"
      value={undefined}
    />
  </div>
  <div
    className="c1"
  >
    <input
      className="c3"
      disabled={true}
      type="submit"
      value="Search"
    />
  </div>
  <div
    className="c1"
  >
    <a
      href="https://doc.bblf.sh/using-babelfish/uast-querying.html"
      rel="noopener noreferrer"
      target="_blank"
    >
      Help
    </a>
  </div>
</form>
`;

exports[`renders correctly with value 1`] = `
.c0 {
  padding: 0.5rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #f9f7f5;
  border-bottom: 1px solid #dad7d0;
  min-height: 2.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 + .c1 {
  margin-left: 1rem;
}

.c2 {
  width: 100%;
  background: transparent;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: 1px solid #dad7d0;
  outline: none;
  font-size: 0.8rem;
  padding: 0.1rem 0.5rem;
}

.c3 {
  padding: 0.5em 0.5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  text-transform: uppercase;
  font-size: 0.8rem;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  border: none;
}

.c3[disabled] {
  opacity: 0.6;
  pointer-events: none;
}

.c3:hover {
  box-shadow: 0 5px 15px rgba(138,128,115,0.8);
}

<form
  className="c0"
  onSubmit={[Function]}
>
  <div
    className="c1"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <input
      className="c2"
      disabled={undefined}
      name="query"
      onChange={[Function]}
      placeholder="UAST Query"
      type="text"
      value="test"
    />
  </div>
  <div
    className="c1"
  >
    <input
      className="c3"
      disabled={false}
      type="submit"
      value="Search"
    />
  </div>
  <div
    className="c1"
  >
    <a
      href="https://doc.bblf.sh/using-babelfish/uast-querying.html"
      rel="noopener noreferrer"
      target="_blank"
    >
      Help
    </a>
  </div>
</form>
`;
